op-node/p2p/discovery.go

```diff
@@ -417,6 +425,18 @@ func (n *NodeP2P) DiscoveryProcess(ctx context.Context, log log.Logger, cfg *rol
 	}
 }
 
+// recordPeerMD saves extended metadata about a peer to the peerstore.
+// if any of the metadata fails to be saved, it is logged and ignored.
+func recordPeerMD(pstore peerstore.Peerstore, id peer.ID, md map[string]interface{}) {
+	for k, v := range md {
+		// if the type of the value being added is not a basic type, then we need to register it with gob
+		// that should be done in a singular place. Since we don't yet load any complex types, we don't need to register any.
+		if err := pstore.Put(id, k, v); err != nil {
```

Comment:
I just realized the `/peers/metadata"` part of the peerstore does not have automatic pruning / TTL like the rest of the peerstore attributes have, and so the metadata here would persist indefinitely. If we, for some reason, discover lots and lots of peers, then this might cause the DB to grow too large.

Reply:
Fair point, especially regarding it never being pruned -- how did you determine that btw?

The storage of ENR and ChainID would be about 224 bytes per peer (not including keys/overhead), so 1MB of storage per ~4k peers.

I'll read through the `store.extendedStore` and friends per your suggestion. Thanks!